next/src/validation/object.ts

```diff
@@ -0,0 +1,34 @@
+import type { ValidationError } from '../form'
+import type { NonBooleanJsfSchema, SchemaValue } from '../types'
+import type { StringValidationErrorType } from './string'
+import { validateSchema } from './schema'
+
+export type ObjectValidationErrorType = StringValidationErrorType
+
+/**
+ * Validate an object against a schema
+ * @param value - The value to validate
+ * @param schema - The schema to validate against
+ * @returns An array of validation errors
+ * @description
+ * Validates each property of object against the schema while keeping track of the path to the property.
+ * Each property is validated with `validateSchema`.
+ */
+export function validateObject(value: SchemaValue, schema: NonBooleanJsfSchema): ValidationError[] {
+  if (typeof schema === 'object' && schema.properties && typeof value === 'object') {
```

Comment:
@lukad / @dragidavid can we adopt the "early return" pattern anywhere we can? In this case it would be:

```js
if (typeof schema !== 'object' || !schema.properties || typeof value !== 'object') {
  return []
}

// the actual code...

```